### PR TITLE
Correct bug no forms in the select contact source campaign when the forms have identical names

### PR DESCRIPTION
**Pull request by tomekkowalczyk**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [- ]
| Deprecations?                          | [- ]
| BC breaks? (use the c.x branch)        | [-]
| Automated tests included?              | [- ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
When we have defined the same form names 
image

and create campaigns on the basis of contact source, only the forms with the highest ID are displayed in the campaign source step. 

in the case of a predefined form, e.g. id 13 after creating the form, e.g. ID 14 with the same name, the field for the campaign is empty

image

The fix adds an identifier to the form name and allows you to display and select all campaign forms
image

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10717"></a>

### CODE REVIEW

File: app/bundles/CampaignBundle/Form/Type/CampaignLeadSourceType.php [hunk 1]
```diff
@@ -27,14 +27,18 @@ class CampaignLeadSourceType extends AbstractType
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $sourceType = $options['data']['sourceType'];
+        $sourceChoices = $options['source_choices'] ?? null;
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Form/Type/CampaignLeadSourceType.php]: The default value should be an array as we iterate on it later on.
```suggestion
        $sourceChoices = $options['source_choices'] ?? [];
```

File: app/bundles/CampaignBundle/Form/Type/CampaignLeadSourceType.php [hunk 2]
```diff
@@ -27,14 +27,18 @@ class CampaignLeadSourceType extends AbstractType
...
+        foreach($sourceChoices as $key => $val) {
+            $sourceChoices[$key] = $val.' #'.$key;
+        }
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Form/Type/CampaignLeadSourceType.php]: I would use parentheses to display the ID as I think that's already done in some other select box.
```suggestion
        foreach($sourceChoices as $key => $val) {
            $sourceChoices[$key] = $val.' ('.$key.')';
        }
```

**Review by escopecz — CHANGES_REQUESTED** [review]
Nice work! I can replicate the issue and this code will fix it. I suggested a couple improvements.

Every PR must have a test. This would be nicely covered by a functional test. Something like this:
```php
<?php

declare(strict_types=1);

namespace Mautic\CampaignBundle\Tests\Controller;

use Mautic\CoreBundle\Test\MauticMysqlTestCase;
use Mautic\FormBundle\Entity\Form;

use function GuzzleHttp\json_decode;

final class SourceControllerTest extends MauticMysqlTestCase
{
    public function testTwoSoucesWithSameName(): void
    {
        $form1 = new Form();
        $form1->setName('test');
        $form1->setFormType('campaign');

        $form2 = new Form();
        $form2->setName('test');
        $form2->setFormType('campaign');

        $this->em->persist($form1);
        $this->em->persist($form2);

        $this->em->flush();
        $this->em->clear();

        $this->client->request('GET', '/s/campaigns/sources/new/random_object_id?sourceType=forms');
        $clientResponse = $this->client->getResponse();
        $this->assertSame(200, $clientResponse->getStatusCode(), $this->client->getResponse()->getContent());

        $html = json_decode($clientResponse->getContent(), true)['newContent'];
        $this->assertStringContainsString("<option value=\"{$form1->getId()}\">test</option></select>", $html);
        $this->assertStringContainsString("<option value=\"{$form2->getId()}\">test</option></select>", $html);
    }
}
```
I did not run the test but it should be close. Just paste it to a new file in the `app/bundles/CampaignBundle/Tests/Controller/SourceControllerTest.php` file, make sure you have `.env` file configured properly so it would run the functional tests on a new database and then run `composer test -- --filter SourceControllerTest` so you could see if the tests passes or not. Let me know if you have any questions!

**Comment by escopecz** [comment]
@tomekkowalczyk thanks for pushing the test in! I polished it so it actually works and it discovered a bug. This is perfect example why each code change needs to have test coverage!

**Review by escopecz — APPROVED** [review]

**Review by RCheesley — APPROVED** [review]
Confirmed the bug where only one of the forms was showing in the dropdown, after applying the PR both show in addition to the form ID. Thanks for the PR @tomekkowalczyk - LGTM! 🚀

**Comment by RCheesley** [comment]
@all-contributors please add @tomekkowalczyk for code

**Pull request merged**